### PR TITLE
Archive File links to basic search (peak groups and data) fixed

### LINKED ISSUE #843: Links to data on Archive Files list page are broken
### Problem

The links to Peak Group Data and Peak Data on the Archive Files list page are broken.

### Steps to reproduce

1. Click on "Archive Files" in the left nav bar
2. Click on "Peak Group Data" or "Peak Data" in one of the rows

### Current behavior

```
Page not found (404)
'Ambiguous model instance [ArchiveFile].  Must specify one of [PeakAnnotationFile,MZFile,RAWFile].'
Request Method:	GET
Request URL:	http://127.0.0.1:8000/DataRepo/search_basic/ArchiveFile/id/iexact/1/peakgroups/
Raised by:	DataRepo.views.search.basic.search_basic
Using the URLconf defined in TraceBase.urls, Django tried these URL patterns, in this order:

admin/
[name='home']
DataRepo/ [name='home']
DataRepo/ upload [name='upload']
DataRepo/ validate [name='validate']
DataRepo/ validatedown [name='validatedown']
DataRepo/ search_basic/<str:mdl>/<str:fld>/<str:cmp>/<str:val>/<str:fmt>/ [name='search_basic']
The current path, DataRepo/search_basic/ArchiveFile/id/iexact/1/peakgroups/, matched the last one.
```

### Expected behavior

A list of data associated with the peak annotation file

### Suggested Change

Might be useful to create a separate list view for archive files of `data_type` `ms_peak_annotation` and `ms_data`.

### Comment

None

-----

### Assumptions

1. List of assumptions made WRT the code
2. E.g. We will assume input is correct (explaining why there is no validation)

### Limitations

1. A list of things this work will specifically not do
2. E.g. This feature will only handle the most frequent use case X

### Affected Components

- change: File path or DB table ...
- add: Environment variable or server setting
- delete: External executable or cron job

### Requirements

- [ ] 1. List of numbered conditions to be met for the feature
- [ ] 2. E.g. Every column/row must display a value, i.e. cannot be empty
- [ ] 3. Numbers for reference & checkboxes for progress tracking

#### GUI Change description

None provided

#### Code Change Description

None provided

#### Tests

- [ ] 1. A description of at least one test for each requirement above.
- [ ] 2. E.g. Test for req 2 that there's an exception when display value is ''
- [ ] 3. Numbers for reference & checkboxes for progress tracking

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

Fixed a bug in the advanced and basic search code then updated the archive file templates.

We had never linked using a model that is part of a composite view where that model had multiple "instances" before.  Doing so with `ArchiveFile` records revealed the bug.  Models that have multiple different ways it links to a composite view (e.g. the peak groups "format"), e.g. via a peak annotation file, an mzXML file, or even a RAW file.

Note, none of the links in the view had the name "ArchiveFile", so linking to the basic search using that model name correctly reported the error "Ambiguous model instance" because there were 3 different instances that you could search on.

But when you change it to any of the "correct" instance names, you still get an error.  That's because the code was supplying the model instance name ("PeakAnnotationFile", "RAWFile", or "MZFile") to the `get_model_by_name` function.

I added a new method `getModelByFromInstance` and supplied the result to `get_model_by_name`.

## Affected Issues/Pull Requests

- Resolves #843

## Review Notes

See comments in-line.

## Checklist

This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    
  - Review period: 2 days 
- Associated issue/pull request requirements:
  
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [ ] [Updated *Unreleased* section of `changelog.md` *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

### CODE REVIEW

File: DataRepo/formats/dataformat_group.py [hunk 1]
```diff
@@ -398,7 +404,7 @@ def meetsAllConditionsByValList(self, fmt, rootrec, query, field_order):
             rootrec, query, field_order
         )
 
-    def searchFieldToDisplayField(self, mdl, fld, val, qry):
+    def searchFieldToDisplayField(self, mdl_instance, fld, val, qry):
```
Comment (hepcat72) [inline DataRepo/formats/dataformat_group.py:407]: C. Tried to disambiguate the argument.  There are multiple places in the code that use "mdl" interchangeably with a "model instance".  E.g. "PeakAnnotationFile" is an "instance" of "ArchiveFile".  This is the mechanism used when constructing a "format" (i.e. composite view composed of fields from multiple models).  Both the PeakGroups and PeakData "formats" have 3 ArchiveFile "instances", each linked to the view via different relational paths.  Searching using one or the other leads to different results.

File: DataRepo/migrations/0034_alter_sample_time_collected.py
Comment (hepcat72) [inline DataRepo/migrations/0034_alter_sample_time_collected.py:1]: C. Whoops.  Leftover file from a different branch.  Ugh.  I'll remove it...

**Review by lparsons — CHANGES_REQUESTED** [review]
This looks good and seems to resolve the issue. There should probably be a test and an update to the CHANGELOG. In case it helps (and to help me understand the new method) I submitted a PR with suggested tests.

**Review by lparsons — APPROVED** [review]
LGTM

**Pull request merged**